File: 2021_summer_algorithm_study_basic/using_cpp/11726.cpp
/*
input :
9

output :
55
*/

#include <iostream>
#include <algorithm>
using namespace std;
using ll = long long;

// 전역 변수 선언
int N;
ll dp[1005];
int MOD = 10007;

ll fibo(int num){
    if (num == 0) return 1;
    if (num == 1) return 1;
    if (dp[num]) return dp[num];

    return dp[num] = ((fibo(num - 1)) % MOD + (fibo(num - 2)) % MOD) % MOD;
}

int main(){
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    freopen("input.txt", "r", stdin);

    cin >> N;
    cout << fibo(N);

    return 0;
}

File: 2021_random_set_self_practice/11726.py
"""
input :
9

output :
55
"""
import sys
sys.stdin = open("input.txt", "r")
input = sys.stdin.readline
MOD = 10007

if __name__ == "__main__":
    dp = [0 for _ in range(1001)]
    dp[1] = 1
    dp[2] = 2
    for i in range(3, 1001):
        dp[i] = (dp[i - 2] + dp[i - 1]) % MOD

    N = int(input())
    print(dp[N] % MOD)
